### DISCUSSION TITLE
Arduino Nano ESP32

### DISCUSSION THREAD

**Discussion by Bjqrn88** [grblHAL/ESP32#101] [Ideas]
Hey All,

Would it be possible to get the grblHAL ESP32 version up and running on a [Arduino Nano ESP32](https://store.arduino.cc/collections/boards-modules/products/nano-esp32-with-headers) on a [Nano Screw Terminal Adapter](https://store.arduino.cc/products/nano-screw-terminal?_pos=2&_psq=ASX00037+OR+ABX00041&_ss=e&_v=1.0&variant=42702175338647)?

best regards

Bjqrn

**Comment by terjeio** [comment]
Yep, should be possible now that S3 is supported (but not yet from the Web Builder). FYI since the S3 has only four RMT channels that will be the max number axes/motors possible, for more currently shift registers (I2S) has to be added.

**Reply by Bjqrn88** [reply]
That would be awesome.. I'm currently running a Arduino Mega 2560 with grbl.. But I would love to get grblHAL up and running on one of my Arduino Nano ESP32's..

**Reply by terjeio** [reply]
Any progress on this?

**Reply by Bjqrn88** [reply]
I don’t have the skills to create a port my self. 🙈

**Reply by terjeio** [reply]
Basically only a map file, [like this](https://github.com/grblHAL/ESP32/blob/master/main/boards/cnc3040_map.h), is needed - which pin does what. I've started one but not yet completed it.

**Reply by Bjqrn88** [reply]
Okay thanks, I will have a look at it later 🫣
